src/Config/ObjectModel/RuntimeConfig.cs

```diff
@@ -374,7 +374,7 @@ public int GetEntityCacheEntryTtl(string entityName)
     public bool CanUseCache()
     {
         bool setSessionContextEnabled = DataSource.GetTypedOptions<MsSqlOptions>()?.SetSessionContext ?? true;
-        return IsCachingEnabled && SqlDataSourceUsed && !setSessionContextEnabled;
+        return IsCachingEnabled && (SqlDataSourceUsed || CosmosDataSourceUsed) && !setSessionContextEnabled;
```

Comment:
`(SqlDataSourceUsed || CosmosDataSourceUsed)` can be removed fully now since all db types allow caching.